torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -125,22 +141,91 @@ std::vector<std::shared_ptr<SugaredValue>> SimpleValue::asTuple(
       [](Value* v) -> std::shared_ptr<SugaredValue> {
     return std::make_shared<SimpleValue>(v);
   };
-  if (value->type()->kind() == TypeKind::TupleType) {
-    auto outputs = createTupleUnpack(value);
+  if (value_->type()->kind() == TypeKind::TupleType) {
+    auto outputs = createTupleUnpack(value_);
     return fmap(outputs, make_simple_value);
-  } else if (value->type()->kind() == TypeKind::ListType) {
+  } else if (value_->type()->kind() == TypeKind::ListType) {
     if (!size_hint) {
       throw ErrorReport(loc)
           << "cannot statically infer the expected size of a list in this context";
     }
-    auto graph = value->owningGraph();
+    auto graph = value_->owningGraph();
     Node* unpack =
-        graph->insertNode(graph->createListUnpack(value, *size_hint));
+        graph->insertNode(graph->createListUnpack(value_, *size_hint));
     return fmap(unpack->outputs(), make_simple_value);
   }
-  throw ErrorReport(loc) << value->type()->str()
+  throw ErrorReport(loc) << value_->type()->str()
                          << " cannot be used as a tuple";
 }
+
+void SimpleValue::setAttr(
+    const SourceRange& loc,
+    Method& m,
+    const std::string& field,
+    Value* newValue,
+    bool shouldDefine) {
```

Comment:
`shouldDefine` is _very_ specific to Classes and not something I would expect to appear in the `setAttr` signature. Another way to organize this would be:

* `setAttr/getAttr` are defined on SimpleValue but never change the members of class.
* A sugared value, `InitializingClassValue` is used for `self` _only for the the init method_, and contains extra logic on setAttr to do the creation.

Reply:
agree. Will follow up in the next PR with this.